shark/src/main/java/shark/HeapAnalyzer.kt

```diff
@@ -86,6 +87,20 @@ class HeapAnalyzer constructor(
       )
     }
 
+    if (obfuscationChecker.isCodeObfuscated() && proguardMapping == null) {
+      val exception = IllegalStateException(
+          """
+Looks like you're running LeakCanary on obfuscated build and proguard mapping is missing. Please
```

Comment:
Maybe make the statement stronger (ie avoid "looks like") and start with what we're finding. Also keep in mind this code can also run in the CLI.

Suggestion:

> The heap dump contains obfuscated class names. The heap analysis is unable to proceed without a mapping file to deobfuscate class names. You can run LeakCanary on obfuscated builds by following the instructions at https://square.github.io/leakcanary/recipes/#using-leakcanary-with-obfuscated-apps